cannot use openapi3.Components{…} (value of type "github.com/getkin/kin-openapi/openapi3".Components) as *"github.com/getkin/kin-openapi/openapi3".Components value in struct literal

Hello,

i wanted to install the SwaggerUI feature from this documentation link: https://goyave.dev/guide/libraries/openapi.html#swaggerui

After installing goyave.dev/openapi3 and setting up the code in my router the following error occures:

```
# goyave.dev/openapi3
..\..\..\..\go\pkg\mod\goyave.dev\openapi3@v0.2.1\openapi.go:46:15: cannot use openapi3.Components{…} (value of type "github.com/getkin/kin-openapi/openapi3".Components) as *"github.com/getkin/kin-openapi/openapi3".Components value in struct literal
```
![image](https://github.com/go-goyave/openapi3/assets/28596891/4e9f6a6b-c724-4563-9a64-3ca0d2f40c3a)

--- a/go.mod
+++ b/go.mod
@@ -3,10 +3,9 @@ module goyave.dev/openapi3
 go 1.16
 
 require (
-	github.com/getkin/kin-openapi v0.95.0
+	github.com/getkin/kin-openapi v0.113.0
 	github.com/go-openapi/swag v0.21.1 // indirect
 	github.com/imdario/mergo v0.3.13
-	github.com/mailru/easyjson v0.7.7 // indirect
 	github.com/stretchr/testify v1.8.1
 	golang.org/x/text v0.4.0
 	goyave.dev/goyave/v4 v4.0.0

--- a/openapi.go
+++ b/openapi.go
@@ -43,7 +43,7 @@ func (g *Generator) Generate(router *goyave.Router) *openapi3.T {
 		},
 		Paths:   make(openapi3.Paths),
 		Servers: makeServers(),
-		Components: openapi3.Components{
+		Components: &openapi3.Components{
 			Schemas:       make(openapi3.Schemas),
 			RequestBodies: make(openapi3.RequestBodies),
 			Responses:     make(openapi3.Responses),

--- a/route_test.go
+++ b/route_test.go
@@ -208,7 +208,7 @@ func (c testController) handler(resp *goyave.Response, req *goyave.Request) {
 }
 
 func (suite *RouteTestSuite) TestGetParamSchema() {
-	spec := &openapi3.T{Components: openapi3.Components{Schemas: openapi3.Schemas{}}}
+	spec := &openapi3.T{Components: &openapi3.Components{Schemas: openapi3.Schemas{}}}
 	refs := NewRefs()
 	router := goyave.NewRouter()
 	route := router.Get("/test/{param}/{id:[0-9]+}/{notint:[a-z0-9]+}", HandlerTest)
@@ -243,7 +243,7 @@ func (suite *RouteTestSuite) TestGetParamSchema() {
 }
 
 func (suite *RouteTestSuite) TestGetParamSchemaCacheAndNaming() {
-	spec := &openapi3.T{Components: openapi3.Components{Schemas: openapi3.Schemas{}}}
+	spec := &openapi3.T{Components: &openapi3.Components{Schemas: openapi3.Schemas{}}}
 	refs := NewRefs()
 	router := goyave.NewRouter()
 	route := router.Get("/{param1:[a-z0-9]+}/{param2:[a-z0-9]+}", HandlerTest)
@@ -265,7 +265,7 @@ func (suite *RouteTestSuite) TestGetParamSchemaCacheAndNaming() {
 
 func (suite *RouteTestSuite) TestConvertPathParameter() {
 	spec := &openapi3.T{
-		Components: openapi3.Components{
+		Components: &openapi3.Components{
 			Schemas:    openapi3.Schemas{},
 			Parameters: openapi3.ParametersMap{},
 		},
@@ -322,7 +322,7 @@ func (suite *RouteTestSuite) TestConvertPathParameter() {
 
 func (suite *RouteTestSuite) TestConvertValidationRules() {
 	spec := &openapi3.T{
-		Components: openapi3.Components{
+		Components: &openapi3.Components{
 			Schemas:    openapi3.Schemas{},
 			Parameters: openapi3.ParametersMap{},
 		},
@@ -362,7 +362,7 @@ func (suite *RouteTestSuite) TestConvertValidationRules() {
 
 func (suite *RouteTestSuite) TestConvertValidationRulesWithBody() {
 	spec := &openapi3.T{
-		Components: openapi3.Components{
+		Components: &openapi3.Components{
 			Schemas:       openapi3.Schemas{},
 			Parameters:    openapi3.ParametersMap{},
 			RequestBodies: openapi3.RequestBodies{},
@@ -411,7 +411,7 @@ func findQueryParamRef(query []*openapi3.ParameterRef, ref string) *openapi3.Par
 
 func (suite *RouteTestSuite) TestConvertOperation() {
 	spec := &openapi3.T{
-		Components: openapi3.Components{
+		Components: &openapi3.Components{
 			Schemas:       openapi3.Schemas{},
 			Parameters:    openapi3.ParametersMap{},
 			RequestBodies: openapi3.RequestBodies{},
@@ -447,7 +447,7 @@ func (suite *RouteTestSuite) TestConvertOperation() {
 
 func (suite *RouteTestSuite) TestConvert() {
 	spec := &openapi3.T{
-		Components: openapi3.Components{
+		Components: &openapi3.Components{
 			Schemas:       openapi3.Schemas{},
 			Parameters:    openapi3.ParametersMap{},
 			RequestBodies: openapi3.RequestBodies{},

--- a/ui_test.go
+++ b/ui_test.go
@@ -51,7 +51,7 @@ const (
     window.onload = function() {
       // Begin Swagger UI call region
       const ui = SwaggerUIBundle({
-        spec: {"components":{},"info":{"title":"Test","version":"0.0.0"},"openapi":"3.0.0","paths":{}},
+        spec: {"info":{"title":"Test","version":"0.0.0"},"openapi":"3.0.0","paths":{}},
         dom_id: '#swagger-ui',
         deepLinking: true,
         presets: [
@@ -83,7 +83,7 @@ func (suite *UITestSuite) TestNewOptions() {
 	}
 	opts := NewUIOptions(spec)
 	suite.NotNil(opts)
-	suite.Equal(`{"components":{},"info":{"title":"Test","version":"0.0.0"},"openapi":"3.0.0","paths":{}}`, opts.Spec)
+	suite.Equal(`{"info":{"title":"Test","version":"0.0.0"},"openapi":"3.0.0","paths":{}}`, opts.Spec)
 	suite.NotEmpty(opts.Favicon16)
 	suite.NotEmpty(opts.Favicon32)
 	suite.NotEmpty(opts.BundleURL)

